src/client/Microsoft.Identity.Client/AppConfig/ConfidentialClientApplicationOptions.cs

```diff
@@ -15,5 +15,22 @@ public class ConfidentialClientApplicationOptions : ApplicationOptions
         /// application registration with PowerShell AzureAD, PowerShell AzureRM, or Azure CLI.
         /// </summary>
         public string ClientSecret { get; set; }
+
+        /// <summary>
+        /// Instructs MSAL.NET to use an Azure regional token service.
+        /// This setting should be set to either the string with the region (preffered) or to 
+        /// "TryAutoDetect" and MSAL.NET will attempt to auto-detect the region. 
+        /// </summary>
+        /// <remarks>
+        /// Region names as per https://docs.microsoft.com/en-us/dotnet/api/microsoft.azure.management.resourcemanager.fluent.core.region?view=azure-dotnet.
+        /// Not all auth flows can use the regional token service. 
+        /// Service To Service (client credential flow) tokens can be obtained from the regional service.
+        /// Requires configuration at the tenant level.
+        /// Auto-detection works on a limited number of Azure artifacts (VMs, Azure functions). 
+        /// If auto-detection fails, the non-regional endpoint will be used.
+        /// If an invalid region name is provided, the non-regional endpoint MIGHT be used or the token request MIGHT fail.
+        /// See https://aka.ms/msal-net-region-discovery for more details.        
+        /// </remarks>
+        public string AzureRegion { get; set; }
```

Comment:
Why can't an empty or null `AzureRegion` mean that auto-detection should be performed?

I feel like currently `AzureRegion` is an overloaded property. It means two things - the region itself and whether detection should be performed or not.

Reply:
customers have say "AutoDetectRegion" or just pass something else, their chosen region. Null will mean there is no region lookup and none provided. because we have `.WithAzureRegion()`, so we need a way to tell if the region is set or not, otherwise we will also do the regional look up when we shouldn't.

Reply:
Yeah, we have to distinguish 3 states: 

- not configured
- configured to default (auto-discovery)
- configured with non-default

It's quite common throughout the library, see Prompt as well. In this case, I think `null` is fine to denote that it was not configured.